data-structures/ds-avltree/c/AVLRouteInC.c

```diff
@@ -0,0 +1,255 @@
+/*
+by Felipe Gabriel
+this code goes through the tree and prints the AVL tree in order, post order and pre order
+this code have all the main functions off a AVL tree 
+*/
+
+#include <stdio.h>
+#include <stdlib.h>
+
+
+typedef struct node{
+	int num;
+	struct node* left;
+	struct node* right;
+}Node;
+
+typedef Node* TreeBin;
+
+TreeBin * newTree(){
+	TreeBin* root = (TreeBin*) malloc(sizeof(TreeBin));
+	if(root)
+	    (*root) = NULL;
+
+	return root;
+}
+
+void clearNode(Node* node){
+	if(!node)	return;
+	clearNode(node->left);
+	clearNode(node->right);
+	free(node);
+	node = NULL;
+}
+
+void clearTree(TreeBin* Tree){
+	if(!Tree)	return;
+	clearNode(*Tree);
+	free(Tree);
+}
+
+int isEmpty(TreeBin* Tree){
+	if(!Tree)	return 1;
+	if(!(*Tree))	return 1;
+	return 0;
+}
+
+int countNode(TreeBin* Tree){
+	int tleft, tright;
+
+	if(isEmpty(Tree))	return 0;
+
+	tleft = countNode(&((*Tree)->left));
+	tright = countNode(&((*Tree)->right));
+
+	return (tleft+tright+1);
+}
+
+int heightTree(TreeBin* Tree){
+	int heightLeft, heightRight;
+
+	if(isEmpty(Tree))	return -1;
+	if(isEmpty(&((*Tree)->left)) && isEmpty(&((*Tree)->right)))	return 0;
+
+	heightLeft = heightTree(&((*Tree)->left));
+	heightRight = heightTree(&((*Tree)->right));
+
+	if(heightLeft > heightRight)	return (heightLeft + 1);
+
+	return (heightRight + 1);
+}
+
+//balancing functions
+// [-1, 1] balanced tree
+// [-2, 2] not balanced tree
+int Balanced(TreeBin *Tree){
+	return heightTree(&((*Tree)->left)) - heightTree(&((*Tree)->right));
+}
+
+void spinRight(TreeBin *Tree){
+	Node *nod;
+	nod = (*Tree)->left;
+	(*Tree)->left = nod->right;
+	nod->right = (*Tree);
+	(*Tree) = nod;
+}
+
+void spinLeft(TreeBin *Tree){
+	Node *nod;
+	nod = (*Tree)->right;
+	(*Tree)->right = nod->left;
+	nod->left = (*Tree);
+	(*Tree) = nod;
+}
+
+void balance(TreeBin *Tree, int n){
+	int eb, it;
```

Comment:
Codacy found an issue: [The scope of the variable 'it' can be reduced.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6412547)